Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/ErrorLogServiceImpl.java

```diff
@@ -71,60 +69,46 @@
 
     @Override
-    public List<LogEventType> search(String service, String exception, Date fromDate, Date toDate) {
+    public List<LogEventType> search(String service, String exception, Date fromDate, Date toDate) throws Exception {
         ListErrorLogRequestMessageType request = new ListErrorLogRequestMessageType();
         request.setConfigAssertion(buildConfigAssertion());
         request.setService(service);
         request.setException(exception);
         request.setFromDate(getXMLGregorianCalendarFrom(fromDate));
         request.setToDate(getXMLGregorianCalendarFrom(toDate));
 
-        try {
-            LogEventSimpleResponseMessageType response = (LogEventSimpleResponseMessageType) invokeClientPort(
-                ADMIN_DASHBOARD_ERRORLOG_LIST, request);
-            logDebug(ADMIN_DASHBOARD_ERRORLOG_LIST, response.getEventLogList().size());
-            return response.getEventLogList();
-        } catch (Exception e) {
-            LOG.error("error during search: {}", e.getLocalizedMessage(), e);
-        }
-        return new ArrayList<>();
+        LogEventSimpleResponseMessageType response = (LogEventSimpleResponseMessageType) invokeClientPort(
+            ADMIN_DASHBOARD_ERRORLOG_LIST, request);
+        logDebug(ADMIN_DASHBOARD_ERRORLOG_LIST, response.getEventLogList().size());
+        return response.getEventLogList();
+
     }
 
     @Override
-    public Map<String, List<String>> getDropdowns() {
+    public Map<String, List<String>> getDropdowns() throws Exception {
         GetSearchFilterRequestMessageType request = new GetSearchFilterRequestMessageType();
         request.setConfigAssertion(buildConfigAssertion());
         Map<String, List<String>> retObj = new HashMap<>();
 
-        try {
-            LogEventSimpleResponseMessageType response = (LogEventSimpleResponseMessageType) invokeClientPort(
-                ADMIN_DASHBOARD_ERRORLOG_GETFILTERS, request);
-            logDebug(ADMIN_DASHBOARD_ERRORLOG_GETFILTERS, response.getExceptionList().size(),
-                response.getServiceList().size());
-            retObj.put(KEY_EXCEPTIONS, response.getExceptionList());
-            retObj.put(KEY_SERVICES, response.getServiceList());
-        } catch (Exception e) {
-            LOG.error("error during get-dropdowns: {}", e.getLocalizedMessage(), e);
-        }
+        LogEventSimpleResponseMessageType response = (LogEventSimpleResponseMessageType) invokeClientPort(
+            ADMIN_DASHBOARD_ERRORLOG_GETFILTERS, request);
+        logDebug(ADMIN_DASHBOARD_ERRORLOG_GETFILTERS, response.getExceptionList().size(),
+            response.getServiceList().size());
+        retObj.put(KEY_EXCEPTIONS, response.getExceptionList());
+        retObj.put(KEY_SERVICES, response.getServiceList());
         return retObj;
     }
 
     @Override
-    public LogEventType getLogEvent(Long id) {
+    public LogEventType getLogEvent(Long id) throws Exception {
```

Comment:
What causes this method to throw an exception apart from Webservice connectivity issues?

Reply:
client.invokePort throw-exception this is old code we may have to refactor later